File: Models/FixVersion.cs
namespace mbm.Models;

public class FixVersion
{
    public int Id { get; set; }
    public int ReleaseId { get; set; }
    public Release? Release { get; set; }
    public string Name { get; set; } = "";
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? JiraFixVersionId { get; set; }
}

File: ClientApp/src/app/core/services/fix-version.ts
import { Service, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { FixVersion } from '../models';

@Service()
export class FixVersionService {
  private readonly http = inject(HttpClient);

  listByRelease(releaseId: number) {
    return this.http.get<FixVersion[]>(`/api/fix-versions?releaseId=${releaseId}`);
  }

  create(releaseId: number, name: string, startDate?: string | null, endDate?: string | null) {
    return this.http.post<FixVersion>('/api/fix-versions', { releaseId, name, startDate, endDate });
  }

  delete(id: number) {
    return this.http.delete<void>(`/api/fix-versions/${id}`);
  }
}
